pkg/resource/alias/hooks.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package alias
+
+import (
+	"context"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+// customUpdateAlias calls UpdateAlias function to update a an alias
+// if it changed.
+func (rm *resourceManager) customUpdateAlias(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateAlias")
+	defer exit(err)
+
+	var updated *resource
+	updated = desired
+	if delta.DifferentAt("Spec.Description") ||
+		delta.DifferentAt("Spec.FunctionName") ||
+		delta.DifferentAt("Spec.FunctionVersion") ||
+		delta.DifferentAt("Spec.RoutingConfig") {
+		updated, err = rm.updateAlias(ctx, updated)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return updated, nil
+}
+
+// updateAlias updates an alias using the AWS API SDK
+func (rm *resourceManager) updateAlias(
+	ctx context.Context,
+	desired *resource,
+) (*resource, error) {
+	var err error
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.updateAlias")
+	defer exit(err)
+
+	input := &svcsdk.UpdateAliasInput{
+		Name:            aws.String(*desired.ko.Spec.Name),
+		FunctionName:    aws.String(*desired.ko.Spec.FunctionName),
+		FunctionVersion: aws.String(*desired.ko.Spec.FunctionVersion),
+		// TODO(hilalymh): We need to figure out when to use revisionId
+		// RevisionId:      aws.String(*desired.ko.Status.RevisionID),
+	}
+	if desired.ko.Spec.Description != nil {
+		input.Description = aws.String(*desired.ko.Spec.Description)
+	} else {
+		input.Description = aws.String("")
+	}
+
+	if desired.ko.Spec.RoutingConfig != nil {
+		input.RoutingConfig = &svcsdk.AliasRoutingConfiguration{
+			AdditionalVersionWeights: desired.ko.Spec.RoutingConfig.AdditionalVersionWeights,
+		}
+	}
+
+	_, err = rm.sdkapi.UpdateAliasWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "UpdateAlias", err)
+	if err != nil {
+		return nil, err
+	}
+	return desired, nil
```

Comment:
what about the non-custom update generated code was not working? in other words, why the need for custom code here?

Reply:
First generation didn't produce the `Update` method... I'm trying to reproduce this.